app/main/helpers/validation_tools.py

```diff
@@ -33,66 +42,77 @@ def question_errors(self, question_id):
                 return
 
         for rule in question_content["validations"]:
-            if not self.test(question, rule["name"]):
+            if not self.test(question_id, question, rule["name"]):
                 return rule["message"]
 
-        # TODO We can't update file URLs until we have the write API,
-        # so uploaded file must have the same extension as the
-        # current one for now
-        current_extension = get_extension(self.service.get(question_id, ''))
-        if get_extension(question.filename) != current_extension:
-            return u"Uploaded file format should be: %s" % current_extension
-
-    def test(self, question, rule):
+    def test(self, question_id, question, rule):
         if not hasattr(self, rule):
             raise ValueError("Validation rule " + rule + " not found")
-        return getattr(self, rule)(question)
+        return getattr(self, rule)(question_id, question)
 
-    def answer_required(self, question):
+    def answer_required(self, question_id, question):
```

Comment:
It's not used in `answer_required`, but it's used in `file_can_be_saved`, and `test` runs every rule the same way